### PR TITLE
Add documentation for Entities and Components in space_shooter_rs 

### PR DESCRIPTION
**Pull request by tigleym**
For #83  

This PR adds a section describing the `Spaceship`, `Enemy`, and `Item` entities and their attached components.

### CODE REVIEW

File: book/src/entities-components.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+There are several
+[entities](https://github.com/amethyst/space_shooter_rs/tree/master/src/entities)
+and
+[components](https://github.com/amethyst/space_shooter_rs/tree/master/src/components)
+in space_shooter_rs. This might make the game's code seem
+intimidating and needlessly complex, but being able to break down functionality
+of an entity into smaller pieces of data will make it much easier to expand upon
+as the game grows.
+
+We will look at three entities present in space_shooter_rs: `Spaceship`
```
Comment (cdsupina) [inline book/src/entities-components.md:16]: I think that referring to components without the word "Component" at the end (as in `ItemComponent`) is fine. However, we should maybe make a note in this book that our naming convention for our custom components is to have "Component" at the end. I also think that now may be the time to make all of the other components in the game, such as `Spaceship` and `Enemy` be consistent with the rest of the new components that we added and change them to `SpaceshipComponent` and `EnemyComponent`. I think that staying consistent is really important to helping new people learn the code.

File: book/src/entities-components.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+*Drawing that groups together components relevant to each entity: Spaceship,
+Enemy, and Item. Made with Excalidraw.*
+
+You might notice that each entity in this example has a component with the same
+name. In space_shooter_rs, functionality that is specific to an entity should
+be contained by its own specific component. For example, the `Item` component
+has properties that are specific to an `Item` entity:
+
+```rust
+pub struct Item {
```
Comment (cdsupina) [inline book/src/entities-components.md:49]:
```suggestion
pub struct ItemComponent {
```
I changed this in #87.

File: book/src/entities-components.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+|Enemy|Data specific to `Enemy` entities|Enemy|
+|Health|Can receive damage|Enemy, Spaceship|
+|Hitbox2D|Can collide in 2D space|Item, Enemy, Spaceship|
+|Item|Data specific to `Item` entities|Item|
+|Motion2D|Can move in 2D space|Item, Enemy, Spaceship|
+|Spaceship|Data specific to `Spaceship` entities|Spaceship|
+|SpriteRender|Is rendered to screen|Item, Enemy, Spaceship|
+|Transform|Has a position, scale, and rotation in the game|Item, Enemy, Spaceship|
+
+Again this seems like a lot of components for only three entities. But if we keep
```
Comment (cdsupina) [inline book/src/entities-components.md:34]:
```suggestion
Again, this seems like a lot of components for only three entities. But if we keep
```

**Review by cdsupina — CHANGES_REQUESTED** [review]
Left some suggestions on grammar. More importantly, I left a comment on the consistency of using the "-Component" suffix at the end of the component names. Let me know what you think.

**Comment by tigleym** [comment]
Thanks @cdsupina for the review! I've addressed your suggestions:

- Made sure all components had "Component" at the end of their name
- Added some pictures for the architecture.md file
- Added extra content on naming conventions for Custom and Tag components

**Review by cdsupina — APPROVED** [review]
Looks great! Go ahead and merge! I love the inclusion of the graphics from our talk at RustFest!

**Pull request merged**